app/controllers/api/v1/metrics_controller.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+#
+# Security: Protect from Forgery.
+#
+module Api
+  module V1
+    class MetricsController < ApplicationController
+      protect_from_forgery with: :null_session
+      
+      def index
+        build_metrics(Metric.all)
+      end
+
+      def create
+        build_metrics(
+          CreateService.perform(metric_params)
+        )
```

Comment:
For better source code readability, I decided to create a separate service which is also beneficial for long-term perspective and reusability.